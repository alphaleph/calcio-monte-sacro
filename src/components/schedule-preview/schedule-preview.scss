@import '../../pages/globals.scss';

.schedule-preview-previous-title {
    margin-bottom: .5rem;
}

.schedule-preview-upcoming-title {
    margin-bottom: .5rem;
}

.schedule-preview-upcoming-cards > * {
    margin: .5rem 0 !important;
}

@media screen and (min-width: 769px) {
    .schedule-preview-upcoming-cards {
        display: flex !important;
        flex-direction: row;
    }

    .schedule-preview-upcoming-cards > * {
        margin: 0 .5rem !important;
    }
}