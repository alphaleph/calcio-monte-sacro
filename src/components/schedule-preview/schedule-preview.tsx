import React from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import { EventCard } from '../event-card/event-card';
import { MarkdownData, MarkdownEventDataNode } from '../../types/index';
import { findMostRecentlyPastIndex } from '../../util';

interface SchedulePreviewOwnProps {
    date?: number;
}

interface SchedulePreviewProps extends SchedulePreviewOwnProps {
    events: MarkdownEventDataNode[];
}

export const PureSchedulePreview = ({
    events,
    date = Date.now(),
}: SchedulePreviewProps) => {
    const mostRecentlyPastIndex = findMostRecentlyPastIndex(events, date);
    const previewScheduleData = [
        events[mostRecentlyPastIndex],
        events[mostRecentlyPastIndex + 1],
        events[mostRecentlyPastIndex + 2],
    ];

    const cardList = previewScheduleData.map((event, index) => (
        <div className="column is-one-third" key={index}>
            <EventCard event={event ? event : null} />
        </div>
    ));

    return (
        <section className="columns is-multiline">
            <div className="column is-one-third">
                <h3 className="has-text-black-main is-size-5">
                    <strong>PREVIOUS EVENT</strong>
                </h3>
            </div>
            <div className="column is-two-thirds">
                <h3 className="has-text-black-main is-size-5">
                    <strong>UPCOMING EVENTS</strong>
                </h3>
            </div>
            {cardList}
        </section>
    );
};

export const SchedulePreview = (props: SchedulePreviewOwnProps) => {
    const data: MarkdownData = useStaticQuery(
        graphql`
            query {
                allMarkdownRemark(
                    sort: { order: ASC, fields: frontmatter___date }
                    filter: { frontmatter: { layout: { eq: "schedule" } } }
                ) {
                    edges {
                        node {
                            frontmatter {
                                title
                                language
                                type
                                venue
                                opposing_team
                                score_ms
                                score_other
                                date
                                layout
                            }
                            fields {
                                slug
                            }
                            html
                        }
                    }
                }
            }
        `
    );
    return (
        <PureSchedulePreview {...props} events={data.allMarkdownRemark.edges} />
    );
};
