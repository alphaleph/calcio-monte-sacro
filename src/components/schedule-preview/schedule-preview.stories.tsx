import React from 'react';
import { storiesOf } from '@storybook/react';
import { PureSchedulePreview } from './schedule-preview';
import { mockScheduleDataNodes } from '../__tests__/test-util';
import '../../pages/index.scss';

storiesOf('SchedulePreview', module).add('default', () => (
    <PureSchedulePreview events={mockScheduleDataNodes} />
));
