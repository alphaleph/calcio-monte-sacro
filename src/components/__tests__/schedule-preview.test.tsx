import React from 'react';
import { SchedulePreview } from '../schedule-preview/schedule-preview';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('Schedule Preview', () => {
    it('renders without crashing', () => {
        const { getByText } = render(<SchedulePreview />);
        expect(getByText('Previous Event')).toBeInTheDocument();
        expect(getByText('Upcoming Events')).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { getByText } = render(<SchedulePreview />);
        expect(getByText('Previous Event')).toBeVisible();
        expect(getByText('Upcoming Events')).toBeVisible();
    });

    it('has not changed', () => {
        const { asFragment } = render(<SchedulePreview />);
        expect(asFragment()).toMatchSnapshot();
    });
});
