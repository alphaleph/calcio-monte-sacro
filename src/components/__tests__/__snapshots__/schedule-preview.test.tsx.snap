// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule Preview has not changed 1`] = `
<DocumentFragment>
  <section
    class="columns is-multiline"
  >
    <div
      class="column is-one-third"
    >
      <h3
        class="has-text-black-main is-size-5"
      >
        <strong>
          PREVIOUS EVENT
        </strong>
      </h3>
    </div>
    <div
      class="column is-two-thirds"
    >
      <h3
        class="has-text-black-main is-size-5"
      >
        <strong>
          UPCOMING EVENTS
        </strong>
      </h3>
    </div>
    <div
      class="column is-one-third"
    >
      <article
        class="event-card-container container"
      >
        <header
          class="event-card-header"
        >
          <p>
            20 NOV 2019
          </p>
          <p
            class="has-text-weight-light"
          >
            Parco Sannazzaro
          </p>
        </header>
        <section
          class="event-card-info has-text-centered columns is-mobile is-vcentered"
        >
          <div
            class="column practice-info"
          >
            <p>
              PRACTICE
            </p>
            <p
              class="has-text-team-holy"
            >
              12:40
            </p>
          </div>
        </section>
      </article>
    </div>
    <div
      class="column is-one-third"
    >
      <article
        class="event-card-container container"
        data-testid="empty"
      />
    </div>
    <div
      class="column is-one-third"
    >
      <article
        class="event-card-container container"
        data-testid="empty"
      />
    </div>
  </section>
</DocumentFragment>
`;
